# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/StyleAnimationValue.h
# Commit: 63603267402e
# Full Hash: 63603267402ee93d6551ef60c35191d99aaee06b
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 3
# Description:
#   Bug 1846516 - [css-properties-values-api] Introduce AnimatedPropertyID/AnimatedPropertIDSet. r=firefox-animation-reviewers,firefox-style-system-reviewers,layout-reviewers,emilio,hiro
#   
#   This will make possible to animate custom properties on Gecko side. For now, the
#   animation code keeps only dealing with nsCSSPropertyID, so behavior is unchanged.
#   
# ==============================================================================

diff -r 753bc39f59ff -r 63603267402e layout/style/StyleAnimationValue.h
--- a/layout/style/StyleAnimationValue.h	Mon Dec 18 09:24:07 2023 +0000
+++ b/layout/style/StyleAnimationValue.h	Mon Dec 18 09:24:08 2023 +0000
@@ -43,6 +43,7 @@
 
 enum class PseudoStyleType : uint8_t;
 struct PropertyStyleAnimationValuePair;
+struct AnimatedPropertyID;
 
 struct AnimationValue {
   explicit AnimationValue(const RefPtr<StyleAnimationValue>& aValue)
@@ -91,12 +92,12 @@
   mozilla::gfx::MatrixScales GetScaleValue(const nsIFrame* aFrame) const;
 
   // Uncompute this AnimationValue and then serialize it.
-  void SerializeSpecifiedValue(nsCSSPropertyID aProperty,
+  void SerializeSpecifiedValue(const AnimatedPropertyID& aProperty,
                                const StylePerDocumentStyleData* aRawData,
                                nsACString& aString) const;
 
   // Check if |*this| and |aToValue| can be interpolated.
-  bool IsInterpolableWith(nsCSSPropertyID aProperty,
+  bool IsInterpolableWith(const AnimatedPropertyID& aProperty,
                           const AnimationValue& aToValue) const;
 
   // Compute the distance between *this and aOther.